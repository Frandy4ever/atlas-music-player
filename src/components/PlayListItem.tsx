import { Play, Pause } from 'lucide-react';

interface PlaylistItemProps {
  title: string;
  artist: string;
  duration: string;
  isPlaying: boolean;
  isPaused: boolean;
  onClick: () => void;
}

const PlaylistItem: React.FC<PlaylistItemProps> = ({
  title,
  artist,
  duration,
  isPlaying,
  isPaused,
  onClick,
}) => {

  return (
    <div
      className={`flex items-center justify-between py-2 px-3 rounded-lg cursor-pointer transition-all group
      ${isPlaying ? "bg-blue-400 text-blue-800 dark:bg-red-400 dark:text-red-800" : "hover:bg-blue-200 dark:hover:bg-red-200 dark:hover:text-white"}
      ${isPaused && !isPlaying ? "bg-gray-400 dark:bg-gray-600" : ""}`}
      onClick={onClick}
    >
      <div>
        <h4 className="text-md font-semibold">{title}</h4>
        <p className="text-sm text-gray-600 group-hover:text-blue-600 dark:text-slate-300 dark:group-hover:text-red-600">
          {artist}
        </p>
      </div>

      <div className="flex items-center gap-3">
        <p className="text-xs">{duration}</p>
        <button className="p-1 rounded-full bg-gray-800 text-white">
          {isPlaying ? <Pause size={18} /> : <Play size={18} />}
        </button>
      </div>
    </div>
  );

};

export default PlaylistItem;
