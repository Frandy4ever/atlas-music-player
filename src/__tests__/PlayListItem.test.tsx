import { render } from "@testing-library/react";
import PlayListItem from "@/components/PlayListItem";

describe("PlayListItem Component", () => {
  it("renders a paused song item", () => {
    const { asFragment } = render(
      <PlayListItem
        title="Test Song"
        artist="Test Artist"
        duration="3:45"
        isPlaying={false}
        isPaused={true}
        onClick={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a playing song item", () => {
    const { asFragment } = render(
      <PlayListItem
        title="Now Playing"
        artist="The Band"
        duration="4:20"
        isPlaying={true}
        isPaused={false}
        onClick={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a normal inactive song item", () => {
    const { asFragment } = render(
      <PlayListItem
        title="Inactive Song"
        artist="No One"
        duration="2:30"
        isPlaying={false}
        isPaused={false}
        onClick={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
