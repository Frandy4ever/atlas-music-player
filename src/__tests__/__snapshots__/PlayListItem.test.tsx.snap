// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component > renders a normal inactive song item 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between py-2 px-3 rounded-lg cursor-pointer transition-all group
      hover:bg-blue-200 dark:hover:bg-red-200 dark:hover:text-white
      "
  >
    <div>
      <h4
        class="text-md font-semibold"
      >
        Inactive Song
      </h4>
      <p
        class="text-sm text-gray-600 group-hover:text-blue-600 dark:text-slate-300 dark:group-hover:text-red-600"
      >
        No One
      </p>
    </div>
    <div
      class="flex items-center gap-3"
    >
      <p
        class="text-xs"
      >
        2:30
      </p>
      <button
        class="p-1 rounded-full bg-gray-800 text-white"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-play"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 5a2 2 0 0 1 3.008-1.728l11.997 6.998a2 2 0 0 1 .003 3.458l-12 7A2 2 0 0 1 5 19z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem Component > renders a paused song item 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between py-2 px-3 rounded-lg cursor-pointer transition-all group
      hover:bg-blue-200 dark:hover:bg-red-200 dark:hover:text-white
      bg-gray-400 dark:bg-gray-600"
  >
    <div>
      <h4
        class="text-md font-semibold"
      >
        Test Song
      </h4>
      <p
        class="text-sm text-gray-600 group-hover:text-blue-600 dark:text-slate-300 dark:group-hover:text-red-600"
      >
        Test Artist
      </p>
    </div>
    <div
      class="flex items-center gap-3"
    >
      <p
        class="text-xs"
      >
        3:45
      </p>
      <button
        class="p-1 rounded-full bg-gray-800 text-white"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-play"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 5a2 2 0 0 1 3.008-1.728l11.997 6.998a2 2 0 0 1 .003 3.458l-12 7A2 2 0 0 1 5 19z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PlayListItem Component > renders a playing song item 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between py-2 px-3 rounded-lg cursor-pointer transition-all group
      bg-blue-400 text-blue-800 dark:bg-red-400 dark:text-red-800
      "
  >
    <div>
      <h4
        class="text-md font-semibold"
      >
        Now Playing
      </h4>
      <p
        class="text-sm text-gray-600 group-hover:text-blue-600 dark:text-slate-300 dark:group-hover:text-red-600"
      >
        The Band
      </p>
    </div>
    <div
      class="flex items-center gap-3"
    >
      <p
        class="text-xs"
      >
        4:20
      </p>
      <button
        class="p-1 rounded-full bg-gray-800 text-white"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-pause"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="1"
            width="5"
            x="14"
            y="3"
          />
          <rect
            height="18"
            rx="1"
            width="5"
            x="5"
            y="3"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
